plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -189,22 +191,37 @@ public function update( &$order ) {
 
 		// Also grab the current status so we can compare.
 		$previous_status = get_post_status( $order->get_id() );
+		// If the order doesn't exist in the DB, we will consider it as new.
+		if ( ! $previous_status ) {
+			$previous_status = 'new';
+		}
 
 		// Update the order.
 		parent::update( $order );
 
-		// Fire a hook depending on the status - this should be considered a creation if it was previously draft status.
-		$new_status = $order->get_status( 'edit' );
+		$current_status = $order->get_status( 'edit' );
 
-		if ( $new_status !== $previous_status && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
-			do_action( 'woocommerce_new_order', $order->get_id(), $order );
-		} else {
-			do_action( 'woocommerce_update_order', $order->get_id(), $order );
+		// We need to remove the wc- prefix from the status for comparison and proper evaluation of new vs updated orders.
+		$previous_status = OrderUtil::remove_status_prefix( $previous_status );
+		$current_status  = OrderUtil::remove_status_prefix( $current_status );
+
+		$non_triggering_order_statuses = array( 'new', 'auto-draft', 'draft', 'checkout-draft', 'refunded', 'failed', 'cancelled' );
```

Comment:
can you add the justification for adding `'refunded', 'failed', 'cancelled'` here, I don't think we were doing this before?

Reply:
Removing these as, after discussion, the conclusion was each order ID should trigger one new order event 👍🏼

Reply:
https://github.com/woocommerce/woocommerce/pull/49098/commits/1175b38586a630bedc6bd969dde03e57adf3a610